examples/advertisement/main.go

```diff
@@ -14,9 +14,18 @@ func main() {
 	must("config adv", adv.Configure(bluetooth.AdvertisementOptions{
 		LocalName: "Go Bluetooth",
 	}))
+	adapter.SetConnectHandler(func(device bluetooth.Addresser, connected bool) {
+		if connected {
+			println("stop advertising")
+			adv.Stop()
+		} else {
+			println("start advertising")
+			adv.Start()
+		}
```

Comment:
I don't quite understand this bit.
This is already happening automatically on nrf chips. The SoftDevice stops advertising on connect and the bluetooth driver enables it again on disconnect (if it was enabled before).
Is this just for Linux, or what is it for?

Reply:
Hm, OK, I did not know SoftDevice would do that automatically. So I've just discovered API and implemented it.

This habit of explicitly disabling advertisement I've learned in some example long time ago and did not really question it. But now, when you asked, it makes me wonder...
Let me verify that really happens, SoftDevice stopping advertisement.

In case SoftDevice indeed does stop advertisements, would you prefer I rollback the example change to avoid confusing people or change it so it stops advertising, say, after 5 mins?

Reply:
>  Let me verify that really happens, SoftDevice stopping advertisement.

Pretty sure the SD stops advertising, otherwise there would be no need for re-enabling it ;)
But I'm not sure about other systems. It's very well possible that BlueZ continues to advertise, for example.

> In case SoftDevice indeed does stop advertisements, would you prefer I rollback the example change to avoid confusing people or change it so it stops advertising, say, after 5 mins?

I think if you want to demonstrate how to stop advertising, stopping after a fixed amount of time is a good way of showing this.

Reply:
Verified advertisement stops after connection is established.

Example adjusted to show how to advertise for 5 minutes and then stop.
Also, it handles "on disconnect" event and resets timer to advertise 5 more minutes after every disconnect.